src/GraphQL/Utilities/Visitors/SchemaValidationVisitor.cs

```diff
@@ -233,6 +238,28 @@ public override void VisitDirectiveArgumentDefinition(QueryArgument argument, Di
             // 4.2
             if (!argument.ResolvedType.IsInputType())
                 throw new InvalidOperationException($"The argument '{argument.Name}' of directive '{type.Name}' must be an input type.");
+
+            // validate default
+            if (argument.DefaultValue is GraphQLValue value)
+            {
+                argument.DefaultValue = Execution.ExecutionHelper.CoerceValue(argument.ResolvedType, Language.CoreToVanillaConverter.Value(value)).Value;
+            }
+            else if (argument.DefaultValue != null && !argument.ResolvedType.IsValidDefault(argument.DefaultValue))
+            {
+                throw new InvalidOperationException($"The default value of argument '{argument.Name}' of directive '{type.Name}' is invalid.");
+            }
```

Comment:
Can be merged into `ValidateQueryArgumentDefaultValue` if change it a little:

`  private void ValidateQueryArgumentDefaultValue(QueryArgument argument, Func<string> error)`

Reply:
For another code in `VisitInputObjectFieldDefinition` it will not be so comfortable so let's keep it as is.